4-WebApp-your-API/4-1-MyOrg/TodoListService/Controllers/TodoListController.cs

```diff
@@ -23,70 +26,160 @@ public class TodoListController : Controller
         public TodoListController(IHttpContextAccessor contextAccessor)
         {
             _contextAccessor = contextAccessor;
+            var test = _contextAccessor.HttpContext.User.GetObjectId();
 
             // Pre-populate with sample data
             if (TodoStore.Count == 0)
             {
-                TodoStore.Add(1, new Todo() { Id = 1, Owner = $"{_contextAccessor.HttpContext.User.Identity.Name}", Title = "Pick up groceries" });
-                TodoStore.Add(2, new Todo() { Id = 2, Owner = $"{_contextAccessor.HttpContext.User.Identity.Name}", Title = "Finish invoice report" });
+                TodoStore.Add(1, new Todo() { Id = 1, Owner = $"{_contextAccessor.HttpContext.User.GetObjectId()}", Title = "Pick up groceries" });
+                TodoStore.Add(2, new Todo() { Id = 2, Owner = $"{_contextAccessor.HttpContext.User.GetObjectId()}", Title = "Finish invoice report" });
+                TodoStore.Add(3, new Todo() { Id = 3, Owner = "Other User", Title = "Rent a car" });
+                TodoStore.Add(4, new Todo() { Id = 4, Owner = "Other User", Title = "Get vaccinated" });
             }
         }
 
-        // GET: api/values
         [HttpGet]
-        [RequiredScope(new string[] { "ToDoList.Read", "ToDoList.Write" })]
+        [RequiredScopeOrAppPermission(
+            AcceptedScope = new string[] { "ToDoList.Read", "ToDoList.ReadWrite" },
```

Comment:
>ToDoList.Read [](http://example.com/codeflow?start=44&length=13)

these strings are used 10s of times in this file, and using them all over like it is is looking poor. Why not declare them as const on top of the class..
const string _todoListReadScope = "ToDoList.Read";
const string _todoListWritePermission = "ToDoList.Read.All"

Reply:
replaced